src/libfmt_macros/lib.rs

```diff
@@ -273,7 +273,11 @@ impl<'a> Parser<'a> {
             }
         } else {
             let msg = format!("expected `{:?}` but string was terminated", c);
-            let pos = self.input.len() + 1; // point at closing `"`
+            // point at closing `"`, unless the last char is `\n` to account for `println`
+            let pos = match self.input.chars().last() {
+                Some('\n') => self.input.len(),
+                _ => self.input.len() + 1,
```

Comment:
This is a hack which will make the span to be shifted one to the left when doing `format!("{\n")`, making the span point at the `\` in stead of the `"`, but the improvement in diagnostics for `println` are worth it given how much of a corner case it is (and we can get back to this later to fix it).